btclightclient/partial_merkle_tree.go

```diff
@@ -0,0 +1,215 @@
+package btclightclient
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"io"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+// / We verify partial merkle tree proof. This logic are using in verifytxoutproof
+// / We are follow the logic in verifytxoutproof.
+
+type PartialMerkleTree struct {
+	numberTransactions uint
+	vBits              []bool
+	vHash              []*chainhash.Hash
+}
+
+type MerkleProof struct {
+	nodeValue  chainhash.Hash
+	merklePath []chainhash.Hash
+	pos        uint32
+}
+
+// TODO: verify this value
+const maxAllowBytes = 65536
+
+// TODO: wrap error
+func readPartialMerkleTree(r io.Reader, buf []byte) (*PartialMerkleTree, error) {
```

Comment:
why do we take `buf` as an argument, rather than creating one in this function?

Reply:
Usually we provide a buffer if we want to reuse to gain performance (to avoid allocations). Here, I don't think we are doing it.

Reply:
I am reusing the btcd code to parse the buffer into readable data. The btcd doesn't have an API for reading 4 bytes from the buffer, so this is the "best" I can do.